### ISSUE TITLE
Compiler crash when accessing member inside auxiliary constructor

### ISSUE THREAD

**Issue by scabug** [scala/bug#10035]
```scala
trait Inner {
  def f(): Outer
}

class Outer(o: Set[Inner]) {
  def this() = this(Set(1).map{
    case k => new Inner {
      def f(): Outer = Outer.this
    }
  })
}
```

```scala

Exception in thread "main" scala.reflect.internal.Types$TypeError: value <none> is not a member of <notype>
	at scala.tools.nsc.typechecker.Contexts$ThrowingReporter.handleError(Contexts.scala:1402)
	at scala.tools.nsc.typechecker.Contexts$ContextReporter.issue(Contexts.scala:1254)
	at scala.tools.nsc.typechecker.Contexts$Context.issue(Contexts.scala:573)
	at scala.tools.nsc.typechecker.Typers$Typer$$anonfun$normalTypedApply$1$1.apply(Typers.scala:4549)
	at scala.tools.nsc.typechecker.Typers$Typer$$anonfun$normalTypedApply$1$1.apply(Typers.scala:4548)
	at scala.tools.nsc.typechecker.Typers$Typer.onError$3(Typers.scala:4522)
	at scala.tools.nsc.typechecker.Typers$Typer.normalTypedApply$1(Typers.scala:4548)
	at scala.tools.nsc.typechecker.Typers$Typer.typedApply$1(Typers.scala:4580)
	at scala.tools.nsc.typechecker.Typers$Typer.typedInAnyMode$1(Typers.scala:5343)

```

**Comment by mrdziuban** [comment]
I'm running into this as a result of a macro I wrote to perform type class derivation. Are there any known solutions or workarounds?

**Comment by som-snytt** [comment]
@mrdziuban 

```
class Outer(var o: Set[Inner]) {
  def this() = {
    this(Set.empty[Inner])
    o = Set(1).map {
      case k => new Inner {
        def f(): Outer = Outer.this
      }
    }
  }
}
```

**Comment by milessabin** [comment]
This appears to be fixed in 2.13.0-M5.

**Comment by Jasper-M** [comment]
It still crashes the REPL for me:

```
error: Internal error: unable to find the outer accessor symbol of <none>
java.lang.StackOverflowError
	at scala.runtime.Statics.anyHash(Statics.java:120)
	at scala.util.hashing.MurmurHash3.listHash(MurmurHash3.scala:169)
	at scala.util.hashing.MurmurHash3$.seqHash(MurmurHash3.scala:220)
	at scala.collection.Seq.hashCode(Seq.scala:38)
	at scala.collection.Seq.hashCode$(Seq.scala:38)
	at scala.collection.AbstractSeq.hashCode(Seq.scala:1102)
	at scala.runtime.Statics.anyHash(Statics.java:120)
	at scala.reflect.internal.util.DefinedPosition.hashCode(Position.scala:81)
	at scala.runtime.Statics.anyHash(Statics.java:120)
	at scala.collection.mutable.HashTable$HashUtils.elemHashCode(HashTable.scala:383)
	at scala.collection.mutable.HashTable$HashUtils.elemHashCode$(HashTable.scala:383)
	at scala.collection.mutable.HashTable.elemHashCode(HashTable.scala:38)
	at scala.collection.mutable.HashTable.findEntry(HashTable.scala:133)
	at scala.collection.mutable.HashMap.get(HashMap.scala:58)
	at scala.collection.mutable.Map$WithDefault.get(Map.scala:191)
	at scala.collection.MapOps.apply(Map.scala:132)
	at scala.collection.MapOps.apply$(Map.scala:132)
	at scala.collection.AbstractMap.apply(Map.scala:343)
	at scala.tools.nsc.reporters.PositionFiltering.testAndLog(PositionFilter.scala:45)
	at scala.tools.nsc.reporters.PositionFiltering.filter(PositionFilter.scala:36)
	at scala.tools.nsc.reporters.PositionFiltering.filter$(PositionFilter.scala:32)
	at scala.tools.nsc.reporters.AbstractReporter.filter(AbstractReporter.scala:17)
	at scala.tools.nsc.reporters.AbstractReporter.info0(AbstractReporter.scala:32)
	at scala.reflect.internal.Reporter.error(Reporting.scala:85)
	at scala.tools.nsc.interpreter.shell.ReplReporterImpl.super$error(Reporter.scala:114)
	at scala.tools.nsc.interpreter.shell.ReplReporterImpl.$anonfun$error$1(Reporter.scala:114)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.tools.nsc.interpreter.shell.ReplReporterImpl.withoutTruncating(Reporter.scala:91)
	at scala.tools.nsc.interpreter.shell.ReplReporterImpl.error(Reporter.scala:114)
	at scala.reflect.internal.Reporting.globalError(Reporting.scala:69)
	at scala.reflect.internal.Reporting.globalError$(Reporting.scala:69)
	at scala.reflect.internal.SymbolTable.globalError(SymbolTable.scala:17)
	at scala.tools.nsc.transform.ExplicitOuter$OuterPathTransformer.outerSelect(ExplicitOuter.scala:235)
	at scala.tools.nsc.transform.ExplicitOuter$OuterPathTransformer.outerPath(ExplicitOuter.scala:267)
	at scala.tools.nsc.transform.ExplicitOuter$OuterPathTransformer.outerPath(ExplicitOuter.scala:267)
	at scala.tools.nsc.transform.ExplicitOuter$OuterPathTransformer.outerPath(ExplicitOuter.scala:267)
	... // you get the gist
```

**Comment by milessabin** [comment]
Ha! Looks like I incidentally fixed it myself ... PR incoming :-D

**Comment by milessabin** [comment]
PR here: https://github.com/scala/scala/pull/7178.

**Comment by timorantalaiho** [comment]
Is there any chance of getting this fix to 2.12?

**Comment by ivanopagano** [comment]
> Is there any chance of getting this fix to 2.12?

Should be in for 2.12.13 unless some regression comes out, I guess @timorantalaiho

**Comment by timorantalaiho** [comment]
> Should be in for 2.12.13 unless some regression comes out, I guess @timorantalaiho

Great! Thankyou!
